docs/architecture.md

```diff
@@ -191,23 +191,31 @@ so their source IP will be rewritten to the Node's IP before going out.
 
 ### ClusterIP Service
 
-<img src="/docs/assets/service_walk.svg.png" width="600" alt="Antrea Service Traffic Walk">
+Antrea supports two ways to implement Services of type ClusterIP - leveraging
+`kube-proxy`; or AntreaProxy that implements load balancing for ClusterIP
+Service traffic with OVS.
+
+In the first way of leveraging `kube-proxy`, Antrea Agent adds OVS flows to
+forward the packets from a Pod to a Service's ClusterIP to the `antrea-gw0`
+port, then `kube-proxy` will intercept the packets and select one Service
+endpoint to be the connection's destination and DNAT the packets to the
+endpoint's IP and port. If the destination endpoint is a local Pod, the packets
+will be forwarded to the Pod directly; if it is on another Node the packets
+will be sent to that Node via the tunnel.
 
-At the moment, Antrea leverages `kube-proxy` to serve traffic for ClusterIP and
-NodePort type Services. The packets from a Pod to a Service's ClusterIP will be
-forwarded through the `antrea-gw0` port, then `kube-proxy` will select one Service
-backend Pod to be the connection's destination and DNAT the packets to the Pod's
-IP and port. If the destination Pod is on the local Node the packets will be
-forwarded to the Pod directly; if it is on another Node the packets will be sent
-to that Node via the tunnel.
+<img src="/docs/assets/service_walk.svg.png" width="600" alt="Antrea Service Traffic Walk">
 
 `kube-proxy` can be used in any supported mode: user-space iptables, or IPVS.
 See the [Kubernetes Service documentation](https://kubernetes.io/docs/concepts/services-networking/service)
 for more details.
 
-In the future, Antrea will also implement ClusterIP Service with OVS, and make
-it a configurable option. This should help achieve better performance than
-`kube-proxy` when `kube-proxy` is used in the user-space or iptables modes.
+When AntreaProxy is enabled, Antrea Agent will add OVS flows that implement
+load balancing and DNAT for the ClusterIP Service traffic. In this way, Service
+traffic load balancing is done inside OVS together with other forwarding, and
+it can achieve better performance than using `kube-proxy`, as there is no extra
+overhead of forwarding Service traffic to host TCP/IP stack and iptables
```

Comment:
host TCP/IP stack is maybe a bit confusing here since AFAIK the TCP implementation is not involved

so maybe: "there is no extra overhead of forwarding the Pod-to-Service traffic to the host's network stack for iptables processing".

Reply:
How about: forwarding the Pod-to-Service traffic to the host's network stack and iptables processing?  I meant overhead from both forwarding and iptables.